tests/cpp_api_parser_test.py

```diff
@@ -41,11 +42,134 @@ def test_enable_if_transform():
         disallow_namespaces=[re.compile("^std$")],
         compiler_flags=["-std=c++17"],
-        input_content=input_content)
+        input_content=input_content,
+    )
 
     output = api_parser.generate_output(config)
 
     entities = list(output["entities"].values())
     assert len(entities) == 1
     requires = entities[0].get("requires")
     assert requires
+
+
+class TestCommentStrip:
+    config = api_parser.Config(
+        input_path="a.cpp",
+    )
+
+    def assert_output(self, expected: str):
+        output = api_parser.generate_output(self.config)
+        doc_strings = [v["doc"]["text"] for v in output.get("entities", {}).values()]
+        assert expected in doc_strings
+
+    def test1(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"//! This is a docstring.",
+                b"int var = 0;",
+            ]
+        )
+        self.assert_output(" This is a docstring.")
+
+    def test2(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"/// This is a docstring.",
+                b"int var = 0;",
+            ]
+        )
+        self.assert_output(" This is a docstring.")
+
+    def test3(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"/** This is a docstring. */",
+                b"int var = 0;",
+            ]
+        )
+        self.assert_output(" This is a docstring.")
+
+    def test4(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"/*! This is a docstring. */",
+                b"int var = 0;",
+            ]
+        )
+        self.assert_output(" This is a docstring.")
+
+    def test5(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"int var = 0; ///< This is a docstring.",
+            ]
+        )
+        self.assert_output(" This is a docstring.")
+
+    def test6(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"/**",
+                b" * This is a docstring.",
+                b" */",
+                b"int var = 0;",
+            ]
+        )
+        self.assert_output("\nThis is a docstring.")
+
+    def test7(self):
+        self.config.input_content = b"\n".join(
+            [
+                b"/*!",
+                b" * This is a docstring.",
+                b" */",
+                b"int var = 0;",
+            ]
+        )
+        self.assert_output("\nThis is a docstring.")
+
+
+def test_function_fields():
+    config = api_parser.Config(
+        input_path="a.cpp",
+        input_content="""
+/// @brief A dummy function for tests.
+///
+/// \\details A more detailed paragraph.
+///
+/// @param arg1 An arg passed by value.
+/// \\param[in] arg2 An unaltered arg passed by reference.
+/// @param[in, out] arg3 An arg passed by reference that gets altered.
+/// @retval int A negative number indicating a specific error, otherwise 0.
```

Comment:
I think to avoid confusion this example should be changed to use `retval` in the way it seems to be intended:

https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-c-function

Specifically, both Sphinx and doxygen seem to intend that the first argument to `retval` is the value being returned, not the type.

Reply:
changed this to
```
/// @retval NULL If unsuccessful.
```

I don't often use this command in doxygen because it doesn't group them into the `@returns` paragraph(s).